openedx/core/djangoapps/programs/utils.py

```diff
@@ -151,7 +184,12 @@ def progress(self, programs=None, count_only=True):
                 if self._is_course_complete(course):
                     completed.append(course)
                 elif self._is_course_in_progress(course):
```

Comment:
Make what's going on here easier to understand by renaming this method `_is_course_enrolled()`, then renaming your `does_course_still_qualify_as_in_progress()` method to `_is_course_in_progress()`. Enrollment alone is no longer enough to determine whether a course is in progress.